llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -2626,8 +2652,8 @@ class VPWidenMemoryRecipe : public VPRecipeBase {
   VPWidenMemoryRecipe(const char unsigned SC, Instruction &I,
                       std::initializer_list<VPValue *> Operands,
                       bool Consecutive, bool Reverse, DebugLoc DL)
-      : VPRecipeBase(SC, Operands, DL), Ingredient(I), Consecutive(Consecutive),
-        Reverse(Reverse) {
+      : VPRecipeBase(SC, Operands, DL), VPIRMetadata(I), Ingredient(I),
```

Comment:
Can this way of initializing `VPIRMetadata(I)` be applied to all recipe constructors that receive their underlying instruction `I` as parameter? This seems like a natural default behavior, could be followed by an update if some other exceptional metadata is desired.

Reply:
Updated for now, although it might be good to encourage avoidance of relying on underlying instructions on construction ;)

Reply:
Certainly, by all means!
Any recipe that can be constructed w/o an underlying instruction should be; at that point its metadata can be passed instead of the instruction itself.